### PR TITLE
feat: flame integration

### PR DESCRIPTION
**Pull request by hdd**
Flame integration, first iteration. 

* runs on tasks
* allows to publish reviewables and thumbnails

### CODE REVIEW

File: projects/framework-flame/extensions/plugins/flame_thumbnail_exporter.py
Comment (lluisFtrack) [inline projects/framework-flame/extensions/plugins/flame_thumbnail_exporter.py:1]: Maybe is not possible or worth it, but I see there is a lot of common stuff between this plugin and the reviewable one, should they be the same and be driven by the options? Just a thought.

**Review by lluisFtrack — CHANGES_REQUESTED** [review]
Please check comments. Also remember to align with the main branch.

File: projects/framework-flame/resource/bootstrap/init.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        tool_name,
+        dialog_name,
+        options,
+    )
+
+
+# https://help.autodesk.com/view/FLAME/2022/ENU/?guid=Flame_API_Python_Hooks_Reference_html
+
+
+def get_timeline_custom_ui_actions():
```
[the remark this answers is not in the document]
Comment (hdd) [reply projects/framework-flame/resource/bootstrap/init.py:30]: Yes, this is how flame works to init menus .

https://help.autodesk.com/view/FLAME/2023/ENU/?guid=Flame_API_Python_Hooks_Reference_html

File: projects/framework-flame/extensions/plugins/flame_scene_opener.py
[the remark this answers is not in the document]
Comment (hdd) [reply projects/framework-flame/extensions/plugins/flame_scene_opener.py:1]: no opener, as there's no way to save a project (is a set of folders, not a file).
I'll cleanup

File: projects/framework-flame/extensions/plugins/flame_thumbnail_exporter.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        component_name = self.options.get('component')
+
+        exported_path = get_temp_path(is_directory=True)
+
+        try:
+            # TODO: thumbnail_path = Export thumbnail
+
+            self.logger.debug(f"Thumbnail has been saved to: {exported_path}.")
+        except Exception as error:
+            raise PluginExecutionError(message=error)
```
[the remark this answers is not in the document]
Comment (hdd) [reply projects/framework-flame/extensions/plugins/flame_thumbnail_exporter.py:31]: this is used.

File: projects/framework-flame/extensions/plugins/flame_thumbnail_exporter.py
Comment (hdd) [reply projects/framework-flame/extensions/plugins/flame_thumbnail_exporter.py:1]: rather keep them separated for clarity

File: projects/framework-flame/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+# Documentation
+
+## Building
+
+### Preparations
+
+
+1. Install Poetry
+
+2. Create a Python 3.7 virtual environment. If you're using an Apple Silicon chip, follow the instructions in the [How to install compatible PySide2 on Silicon based Mac](../../README.md#how-to-install-compatible-pyside2-on-silicon-based-mac) section.
```
[the remark this answers is not in the document]
Comment (hdd) [reply projects/framework-flame/README.md:14]: updated, thanks.

File: projects/framework-flame/resource/bootstrap/init.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+#     return get_ftrack_menu()
+#
+# def get_mediahub_archives_custom_ui_actions():
+#     # Adds custom actions in the MediaHub's Archives browser. There is no selection returned for an archive,
+#     # so the hook can only be used to trigger an operation from the Archive panel. It cannot affect its content.
+#     return get_ftrack_menu()
+#
+# def get_batch_custom_ui_actions():
+#     # Adds custom actions to the contextual menu available in Batch.
+#     return get_ftrack_menu()
```
[the remark this answers is not in the document]
Comment (hdd) [reply projects/framework-flame/resource/bootstrap/init.py]: ended up removing them.

**Comment by hdd** [comment]
@lluisFtrack 
I'm aware the wiretap.py module is not currently used or sourced anywhere.
I'm happy to nuke it if you feel like but could be useful for the next iteration.
let me know.

File: projects/framework-flame/extensions/plugins/flame_thumbnail_exporter.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hdd) [reply projects/framework-flame/extensions/plugins/flame_thumbnail_exporter.py:31]: realised now what you meant, I'll cleanup

**Comment by lluisFtrack** [comment]
> @lluisFtrack I'm aware the wiretap.py module is not currently used or sourced anywhere. I'm happy to nuke it if you feel like but could be useful for the next iteration. let me know.

We are trying to not leave any not used code in the production releases. So I would say nuke it for now and in case is needed, add it in the next iteration.

**Review by lluisFtrack — DISMISSED** [review]
Tested on Mac and works. I think we need documentation for this, at least to explain where the menu is.

I think we will need to check the run in main thread thing sooner than later.

For the rest, all looks good to me, I would remove the wiretap.py module tough before merging.

Cheers.

**Review by lluisFtrack — DISMISSED** [review]

**Review by lluisFtrack — APPROVED** [review]

**Pull request merged**